xtask/src/utils.rs

```diff
@@ -1,5 +1,182 @@
 use serde_json::Value;
-use std::{process::Command, time::Duration};
+use std::{
+    collections::HashMap,
+    path::Path,
+    process::{Child, Command, Stdio},
+    time::Duration,
+};
+
+use crate::{endgroup, group};
+
+// Cargo utils -----------------------------------------------------------
+
+/// Run a cargo command
+pub(crate) fn run_cargo(command: &str, params: Params, envs: HashMap<&str, String>, error: &str) {
+    run_cargo_with_path::<String>(command, params, envs, None, error)
+}
+
+/// Run acargo command with the passed directory as the current directory
+pub(crate) fn run_cargo_with_path<P: AsRef<Path>>(
+    command: &str,
+    params: Params,
+    envs: HashMap<&str, String>,
+    path: Option<P>,
+    error: &str,
+) {
+    info!("cargo {} {}\n", command, params.params.join(" "));
+    let mut cargo = Command::new("cargo");
+    cargo
+        .env("CARGO_INCREMENTAL", "0")
+        .envs(&envs)
+        .arg(command)
+        .args(&params.params)
+        .stdout(Stdio::inherit()) // Send stdout directly to terminal
+        .stderr(Stdio::inherit()); // Send stderr directly to terminal
+
+    if let Some(path) = path {
+        cargo.current_dir(path);
+    }
+
+    // Handle cargo child process
+    let rustup_process = cargo.spawn().expect(error);
```

Comment:
I don't think this is a rustup process all the time, `cargo_process` would be a more accurate name I think.

Reply:
ahah good catch this is a leftover of my previous solution to circumvent the toolchain issue using `rustup run`.